### ISSUE TITLE
Info window content

### ISSUE THREAD

**Issue by mmartinovich** [jmelosegui/GooglemapMvc#99]
Hi,

I am trying to add an info windows to a marker. Everything works except "Title" that should be in info window but is rendered outside google map.
This is the link where you can see the problem:
http://alterpoljoprivreda.com/Home/Mapa

You can see the titles are rendered above map instead in the window.

If you could give any pointers that would be great!

Thank you
Marin

**Comment by jmelosegui** [comment]
Could you share the source code you used to generate that map?

**Comment by mmartinovich** [comment]
@using System.Configuration
@using System.Drawing
@using System.Globalization
@using AlterPop.Models.Organizacija
@using Jmelosegui.Mvc.GoogleMap

@model AlterPop.Models.SharedModels.ProizvodjaciViewModel
@{
    ViewBag.Title = "Mapa";
}

<section class="ct-single-event">


    <div class="ct-single-event-title text-center ct-u-marginTop20">
        <h3><span>Mapa proizvođača</span></h3>
    </div>


            @(Html.GoogleMap()
                .Name("map")
                .Height(500)
                .Culture(CultureInfo.GetCultureInfo("hr-HR"))
                .Center(c => c.Latitude(Model.LatitudeDouble)
                            .Longitude(Model.LongitudeDouble))
                .FitToMarkersBounds(true)
                .EnableMarkersClustering(options =>
                {
                    options.AverageCenter = true;
                    options.ZoomOnClick = true;
                    options.HideSingleGroupMarker = true;
                })

                .BindTo<OrganizacijaViewModel, Marker>
                                (Model.ListaOrganizacija, mappings => mappings.For<OrganizacijaViewModel>
                            (
                                binding => binding.ItemDataBound
                                    (
                                        (marker, organizacija) =>
                                        {
                                            marker.Latitude = organizacija.LatitudeDouble;
                                            marker.Longitude = organizacija.LongitudeDouble;
                                            marker.Title = organizacija.Naziv;
                                            marker.Clickable = true;
                                            marker.Window = new InfoWindow(marker)
                                            {
                                                Template =
                                                {
                                                    Content = () => ViewContext.Writer.Write(organizacija.Naziv)
                                                }
                                            };
                                        }
                                    )
                            )
                        )
            )


    @section Scripts {
        @Scripts.Render("~/bundles/jqueryval")
        @(Html.GoogleMap().ScriptRegistrar())
    }

</section>

**Comment by jmelosegui** [comment]
I found the issue:

In order to render an HTML code, the control uses the class `HtmlTemplate` which in turns uses a `TextWriter` (by accesing the `Writer` property of the `ViewContext`). There is no way to obtain the string representation of the information written by the writer (like for example in an `StringWriter`). That's why you need to use an  statement block `@{  }` and call the `Render()` method of the control inside it, instead of using an expresion `@( )` to render the html content of the `InfoWindow`.

TL;DR;
You should use the following code. Notice I changed the the expresion `@( )` (which returns a string) by the statement block `@{  }` (which renders directly to the view using the `ViewContext.Writer`).

```
@{
            Html.GoogleMap()
                .Name("map")
                .Height(500)
                .Culture(CultureInfo.GetCultureInfo("hr-HR"))
                .Center(c => c.Latitude(44.1148781)
                    .Longitude(15.2289514))
                .FitToMarkersBounds(true)
                .EnableMarkersClustering(options =>
                {
                    options.AverageCenter = true;
                    options.ZoomOnClick = true;
                    options.HideSingleGroupMarker = true;
                })

                .BindTo<OrganizacijaViewModel, Marker>
                (Model, mappings => mappings.For<OrganizacijaViewModel>
                    (
                        binding => binding.ItemDataBound
                            (
                                (marker, organizacija) =>
                                {
                                    marker.Latitude = organizacija.LatitudeDouble;
                                    marker.Longitude = organizacija.LongitudeDouble;
                                    marker.Title = organizacija.Naziv;
                                    marker.Clickable = true;
                                    marker.Window = new InfoWindow(marker)
                                    {
                                        Template =
                                        {
                                            Content = () => ViewContext.Writer.Write(organizacija.Naziv)
                                        }
                                    };
                                }
                            )
                    )
                ).Render();
}
```
I acknowledge this is a bit confusing and maybe I should think in a way to make it easier.
Any way, let me know if this solve your issue.

**Comment by mmartinovich** [comment]
Hi,

Your solution worked!
Thank you for fast answer!

All the best!

**Comment by mmartinovich** [comment]
Hi Juan,

clustering blue icon is missing? I see it is missing from your web site but
it was there yesterday...

thx

**Comment by jmelosegui** [comment]
Yes your right. I am also receiving a 404 trying to reach the images urls (ie: http://google-maps-utility-library-v3.googlecode.com/svn/trunk/markerclusterer/images/m3.png).

Fortunately this control allows you to customize the image for marker clustering (see [this](http://www.jmelosegui.com/map/marker/MarkerClustering) for mor info). 

Unfortunately I have a bug ( #102 ) in the control which is braking the above mentioned sample.

I just fixed (see #103 for more details) and you should be able to use an updated version following [this](https://github.com/jmelosegui/GooglemapMvc#self-service-releases)

**Comment by mmartinovich** [comment]
I used custom icon so I am ok. Thank you again!

**Comment by jmelosegui** [comment]
Nice !!!. Thank you for the feedback.
